### PR TITLE
E2e test with cypress

### PR DESCRIPTION
**Pull request by ElevnLi**

### CODE REVIEW

**Comment by ElevnLi** [comment]
Screen Shot 2021-05-07 at 11 47 56 am

**Comment by ElevnLi** [comment]
This is a version that works for login and upload weedcoco, but I reckon it's not the most elegant way doing it. I tried a bit from https://github.com/testing-library/cypress-testing-library/blob/97939da7d4707a71049884c0324c0eda56e26fc2/cypress/integration/find.spec.js but didn't make it to locate element easily like finding the sign in button even.

File: search/src/Components/auth/auth_prompt.js [hunk 1]
```diff
@@ -49,11 +49,11 @@ export default function AuthPrompt(props) {
     <div>
       <div style={{display: 'flex', flexDirection: 'row', alignItems: 'center'}}>
         Please&nbsp;
-        <Button variant="outlined" color="primary" onClick={handleLogin}>
+        <Button id="sign_in_button" variant="outlined" color="primary" onClick={handleLogin}>
```
Comment (jnothman) [inline search/src/Components/auth/auth_prompt.js:52]: Yeah, so this is definitely the kind of thing where the preferred testing code should _not_ use the ID. Rather, you should use use `cy.findByRole('button', {name: /Sign In/})`.

See https://testing-library.com/docs/queries/byrole/ which mentions the default `role` of buttons and the "accessible name" being the text of the button. See also "Rules of ARIA attribute usage by HTML element" at [the aria spec](https://www.w3.org/TR/html-aria).

Comment (ElevnLi) [reply search/src/Components/auth/auth_prompt.js:52]: It worths trying out for different combination to actually use the api
Screen Shot 2021-05-07 at 1 06 43 pm

Comment (ElevnLi) [reply search/src/Components/auth/auth_prompt.js:52]: Screen Shot 2021-05-07 at 1 10 58 pm

Comment (jnothman) [reply search/src/Components/auth/auth_prompt.js:52]: Strange that there are multiple elements but we can't see them listed...?

**Comment by jnothman** [comment]
There also seems to be a dep version conflict between `node` and `cypress`.

**Review by jnothman — CHANGES_REQUESTED** [review]
Can we please also add the cypress test run to `.github/workflows/frontend.yml`?

**Comment by jnothman** [comment]
This is now complaining that 001_image.png is missing. Did you forget to commit it? (perhaps use a low-quality jpeg to save space??)

**Comment by jnothman** [comment]
I should clarify: it turns out that the `celery` and `django` jobs failed, since `.env` didn't exist. This is now rectified.

#457 improves handling such errors so that they lead to CI failure.
afc40d98 halved the runtime of the frontend.yml check by caching layers in the docker build.

**Comment by jnothman** [comment]
It passes 🥂🥳🎉

**Comment by jnothman** [comment]
Okay to take this out of Draft mode?

File: search/cypress/integration/weedai_upload.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+describe('Test the overallworkflow', () => {
```
Comment (jnothman) [inline search/cypress/integration/weedai_upload.js:1]:
```suggestion
describe('Test the overall upload workflow', () => {
```

**Review by jnothman — APPROVED** [review]
lgtm

**Review by jnothman — APPROVED** [review]
Awesome refactor, thanks

**Pull request merged**